src/core/dom/query.js

```diff
@@ -331,3 +331,64 @@ export function elementByTag(element, tagName) {
   assertIsName(tagName);
   return element./*OK*/ querySelector(tagName);
 }
+
+/**
+ * Returns the original nodes of the custom element without any service
+ * nodes that could have been added for markup. These nodes can include
+ * Text, Comment and other child nodes.
+ *
+ * @param {!Node} element
+ * @return {!Array<!Node>}
+ */
+export function getRealChildNodes(element) {
+  return childNodes(element, (node) => !isInternalOrServiceNode(node));
+}
+
+/**
+ * Returns the original children of the custom element without any service
+ * nodes that could have been added for markup.
+ *
+ * @param {!Element} element
+ * @return {!Array<!Element>}
+ * @package @final
+ */
+export function getRealChildren(element) {
```

Comment:
I initially felt like this should go directly under `#core/dom`, but the more I look at it the more I think this was the right move. It diverges from the convention of functions in this file though (ie. we have `childNodes` rather than `getChildNodes`). Can we rename these to maintain consistency?
- `getRealChildNodes` -> `realChildNodes` (deferring to `childNodes`)
- `getRealChildren` -> `realChildElements` (deferring to `childElements`)

Since the identifiers are pretty unique, I suspect it would be an easy find/replace to make this change

Reply:
Done